articles/openshift/oshift-how-build-app-private-repo.md

```diff
@@ -0,0 +1,119 @@
+---
+title: How to build an OpenShift application from a private Github repo | UKCloud Ltd
+description: Shows how to build an OpenShift application where the source code is in a "private" github repository secured with an SSH Key Pair.
+services: openshift
+author: Andy Watkins
+
+toc_rootlink: How To
+toc_sub1:
+toc_sub2:
+toc_sub3:
+toc_sub4:
+toc_title: Build an OpenShift app from a private Github repo
+toc_fullpath: How To/oshift-how-build-app-private-repo.md
+toc_mdlink: oshift-how-build-app-private-repo.md
+---
+
+# How to build an OpenShift application from a private Github repo
+
+This guide assumes you have access to the command-line `oc` client, and have
+logged in to your openshift instance with `oc login`
+
+With a private Github repo, you want to ensure it stays hidden from prying
+eyes, but you also want your Openshift app to be able to build from the repo.
+The way around this is to use a 
+[Deploy key](https://developer.github.com/v3/guides/managing-deploy-keys/#deploy-keys).
+
+
+### Steps:
+**First**
+create an ssh key-pair (don't use your regular one, and make sure
+you don't overwrite your regular one either!!)
+ 
+```bash
+ssh-keygen -t rsa -b 4096 -c "jbloggs@mycompany.com" -f my_github_deploy_key
+```
+This will generate both the private and public key files. The public one will
```

Comment:
Make sure there's an empty line between the end of the code sample and the next paragraph.
Use a tab to indent this paragraph within the numbered step.

Reply:
Hint: 4 spaces seems to work as well as a tab (most software devs have their editors configured to use spaces instead of tabs these days - particularly Python devs)

Reply:
I'll make sure any guidelines for tabs also mention 4 spaces as an alternative, thanks.